Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
+        // already promisified
+        Document : processDocument,
+        // already promisified
+        Folder : processFolder,
+        Placemark : deferAndWrapInPromise(processPlacemark),
+        NetworkLink : deferAndWrapInPromise(processNetworkLink),
+        GroundOverlay : deferAndWrapInPromise(processGroundOverlay),
+        PhotoOverlay : deferAndWrapInPromise(processUnsupported),
+        ScreenOverlay : deferAndWrapInPromise(processUnsupported),
+        Tour : deferAndWrapInPromise(processUnsupported)
+    };
+
+    function deferAndWrapInPromise(originalFunction){
+        return function(){
+            var deferred = when.defer();
+            var args = Array.prototype.slice.call(arguments);
+            var context = this;
+            setTimeout(function deferredFunc(){
```

Comment:
Hmm, I only used `setTimeout` because the current version of when.js in cesium didn't support setting up stuff asynchronously (at least I couldn't tell in my research).  A more updated promise lib that was a+ compatible might do things in a more performant way without having to burden the render loop with non-rendering stuff.  I could always try stuffing bluebird in there and profiling and comparing to when.js to see if it makes a difference.  Anything to look for as a tell for `setTimeout` being slow? 

That said, timeboxing `update` calls might be a big win anyway.  With very large kml files, we see very long calls to `update` as cesium processes the initial entities produced by the giant kml we use.  

I'll wait for @shunter before doing anything now.